rotkehlchen/chain/evm/decoding/xdai_bridge/decoder.py

```diff
@@ -0,0 +1,125 @@
+
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.chain.ethereum.utils import asset_normalized_value
+from rotkehlchen.chain.evm.decoding.constants import CPT_XDAI_BRIDGE, XDAI_BRIDGE_CPT_DETAILS
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails, EventCategory
+from rotkehlchen.chain.evm.decoding.utils import bridge_match_transfer, bridge_prepare_data
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChainID, ChecksumEvmAddress, DecoderEventMappingType
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
+
+if TYPE_CHECKING:
+    from rotkehlchen.assets.asset import Asset
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class XdaiBridgeCommonDecoder(DecoderInterface, metaclass=abc.ABCMeta):
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+            deposit_topic: bytes,
+            withdrawal_topic: Optional[bytes],  # withdrawal is currently unsupported on gnosis
+            bridge_address: ChecksumEvmAddress,
+            bridged_asset: 'Asset',
+            source_chain: ChainID,
+            target_chain: ChainID,
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.bridge_address = bridge_address
+        self.bridged_asset = bridged_asset
+        self.bridged_crypto_asset = bridged_asset.resolve_to_crypto_asset()
+        self.bridged_asset_with_symbol = bridged_asset.resolve_to_asset_with_symbol()
+        self.deposit_topic = deposit_topic
+        self.withdrawal_topic = withdrawal_topic
+        self.source_chain = source_chain
+        self.target_chain = target_chain
+
+    def _decode_bridged_asset(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes a bridging event for the `bridged_asset`, either a deposit or a withdrawal."""
+        if context.tx_log.topics[0] == self.deposit_topic:
+            from_address = context.transaction.from_address
+            to_address = self.bridge_address
+        elif context.tx_log.topics[0] == self.withdrawal_topic:
+            from_address = self.bridge_address
+            to_address = hex_or_bytes_to_address(context.tx_log.data[0:32])
+        else:
+            return DEFAULT_DECODING_OUTPUT
```

Comment:
You need empty line here to separate code areas

```suggestion
            return DEFAULT_DECODING_OUTPUT

```